### PR TITLE
Fixed where strokes would not appear under certain conditions (related #1686)

### PR DESCRIPTION
**Pull request by takeshi-1000**
https://github.com/airbnb/lottie-ios/pull/1686 has some incorrect logic (As pointed out at https://github.com/airbnb/lottie-ios/pull/1686#issuecomment-1201896213),
This PR fix the logic not to degrate.

It also supports core animation logic, and test seems successful in my envrionment.

### CODE REVIEW

**Comment by takeshi-1000** [comment]
I don't know the cause of the linting failure at first glance, so I'll check it when I have time.

File: Sources/Private/CoreAnimation/Animations/StrokeAnimation.swift [hunk 1]
```diff
@@ -59,6 +59,9 @@ extension CAShapeLayer {
         try KeyframeGroup(keyframes: $0)
           .exactlyOneKeyframe(context: context, description: "stroke dashPattern").value.cgFloatValue as NSNumber
       }
+      if lineDashPattern?.allSatisfy({ $0.floatValue.isZero }) ?? false {
```
Comment (calda) [inline Sources/Private/CoreAnimation/Animations/StrokeAnimation.swift:62]: I personally find `== true` more idiomatic than `?? false` (subjective for sure though!)

```suggestion
      if linePattern.allSatisfy({ $0.floatValue.isZero }) == true {
```

**Review by calda — APPROVED** [review]
Thanks! To fix the lint failure just run `bundle exec rake format:swift` or `swift package --allow-writing-to-package-directory format` (if you don't have ruby bundler installed).

File: Sources/Private/MainThread/NodeRenderSystem/Nodes/RenderNodes/GradientStrokeNode.swift [hunk 2]
```diff
@@ -140,7 +140,7 @@ final class GradientStrokeNode: AnimatorNode, RenderNode {
 
     /// Get dash lengths
     let dashLengths = strokeProperties.dashPattern.value.map { $0.cgFloatValue }
-    if dashLengths.count > 0 {
+    if dashLengths.count > 0 && !dashLengths.allSatisfy({ $0.isZero }) {
```
Comment (calda) [inline Sources/Private/MainThread/NodeRenderSystem/Nodes/RenderNodes/GradientStrokeNode.swift:143]: Looking at the lint output I think it wants this to be:

```suggestion
    if dashLengths.count > 0, !dashLengths.allSatisfy({ $0.isZero }) {
```

File: Sources/Private/MainThread/NodeRenderSystem/Nodes/RenderNodes/StrokeNode.swift [hunk 3]
```diff
@@ -118,7 +118,7 @@ final class StrokeNode: AnimatorNode, RenderNode {
 
     /// Get dash lengths
     let dashLengths = strokeProperties.dashPattern.value.map { $0.cgFloatValue }
-    if dashLengths.count > 0 {
+    if dashLengths.count > 0 && !dashLengths.allSatisfy({ $0.isZero }) {
```
Comment (calda) [inline Sources/Private/MainThread/NodeRenderSystem/Nodes/RenderNodes/StrokeNode.swift:121]: Same as https://github.com/airbnb/lottie-ios/pull/1691/files#r936164838:

```suggestion
    if dashLengths.count > 0, !dashLengths.allSatisfy({ $0.isZero }) {
```

File: Sources/Private/CoreAnimation/Animations/StrokeAnimation.swift (hunk 1 above)
Comment (takeshi-1000) [reply Sources/Private/CoreAnimation/Animations/StrokeAnimation.swift]: exactily, I changed 👍

**Comment by takeshi-1000** [comment]
@calda 
Thank you for reviewing! I change some code, please check.

**Comment by calda** [comment]
Thanks!

**Pull request merged**